Targets/Mashow/Sources/DrinkSelection/View/DrinkSelectionViewController.swift

```diff
@@ -0,0 +1,287 @@
+    
+    init(viewModel: DrinkSelectionViewModel!) {
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    let viewModel: DrinkSelectionViewModel!
+    private var cancellables = Set<AnyCancellable>()
+    private let drinkTypeList = DrinkSelectionViewModel.DrinkType.allCases
+    
+    private var removedButtonTag: Int?
+    typealias TypeButtonTag = Int
+    private var addedDrinkTypeButtons = [TypeButtonTag: UIButton]()
+    
+    private let pageViewController = UIPageViewController(
+        transitionStyle: .scroll,
+        navigationOrientation: .horizontal
+    )
+    
+    private lazy var leftArrowButton: UIButton = {
+        let button = UIButton()
+        button.setImage(UIImage(systemName: "arrowtriangle.left.fill"), for: .normal)
+        button.tintColor = .white
+        return button
+    }()
+    
+    var backgroundView = UIImageView()
+    
+    private lazy var rightArrowButton: UIButton = {
+        let button = UIButton()
+        button.setImage(UIImage(systemName: "arrowtriangle.right.fill"), for: .normal)
+        button.tintColor = .white
+        return button
+    }()
+    
+    private lazy var addedTypesStackView: UIStackView = {
+        let stack = UIStackView()
+        stack.axis = .horizontal
+        stack.spacing = 4
+        stack.alignment = .center
+        return stack
+    }()
+    
+    private lazy var bottomNextButton: UIButton = {
+        let button = UIButton()
+        button.setTitle("다음", for: .normal)
+        button.titleLabel?.font = .pretendard(size: 20, weight: .bold)
+        button.tintColor = .white
+        button.layer.cornerRadius = 13
+        button.backgroundColor = UIColor.clear
+        button.layer.masksToBounds = true
+        
+        let blurEffect = UIBlurEffect(style: .systemUltraThinMaterialDark)
+        let blurEffectView = UIVisualEffectView(effect: blurEffect)
+        blurEffectView.layer.cornerRadius = 13
+        blurEffectView.layer.masksToBounds = true
+        blurEffectView.isUserInteractionEnabled = false
+        blurEffectView.translatesAutoresizingMaskIntoConstraints = false
+        button.insertSubview(blurEffectView, at: 0)
+        
+        NSLayoutConstraint.activate([
+            blurEffectView.leadingAnchor.constraint(equalTo: button.leadingAnchor),
+            blurEffectView.trailingAnchor.constraint(equalTo: button.trailingAnchor),
+            blurEffectView.topAnchor.constraint(equalTo: button.topAnchor),
+            blurEffectView.bottomAnchor.constraint(equalTo: button.bottomAnchor)
+        ])
+        return button
+    }()
+    
+    // MARK: - Lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        bind()
+        setupNavigationBar()
+        setupLayouts()
+        setupHandlers()
+    }
+}
+
+private extension DrinkSelectionViewController {
```

Comment:
```suggestion
extension DrinkSelectionViewController {
```
메서드에 있는 private을 떼도 되는데 아무래도 이 쪽이 더 쉬울듯ㅋㅋ